snapcraft/application.py

```diff
@@ -150,16 +150,6 @@ def _configure_services(self, provider_name: str | None) -> None:
 
         super()._configure_services(provider_name)
 
-    @property
-    def command_groups(self):
-        """Replace craft-application's LifecycleCommand group."""
-        _command_groups = super().command_groups
-        for index, command_group in enumerate(_command_groups):
-            if command_group.name == "Lifecycle":
-                _command_groups[index] = cli.CORE24_LIFECYCLE_COMMAND_GROUP
-
-        return _command_groups
```

Comment:
I would have had to do more hacks for `InitCommand`, hence https://github.com/canonical/craft-application/pull/545.